src/modules/75-cf/pages/pipeline-studio/views/StageOverview/__tests__/StageOverview.test.tsx

```diff
@@ -0,0 +1,175 @@
+  getPipelinePromise: jest.fn().mockImplementation(() => Promise.resolve(PipelineResponse)),
+  putPipelinePromise: jest.fn().mockImplementation(() => Promise.resolve({ status: 'SUCCESS' })),
+  createPipelinePromise: jest.fn().mockImplementation(() => Promise.resolve({ status: 'SUCCESS' })),
+  useCreateVariablesV2: jest.fn(() => ({ mutate: jest.fn(() => Promise.resolve({ data: { yaml: '' } })) })),
+  useGetSchemaYaml: jest.fn(() => ({})),
+  useGetStepYamlSchema: jest.fn(() => ({})),
+  useGetInputsetYaml: jest.fn(() => ({ data: null }))
+}))
+
+jest.mock('@common/hooks', () => ({
+  ...(jest.requireActual('@common/hooks') as any),
+  useMutateAsGet: jest.fn()
+}))
+
+jest.mock('@wings-software/uicore', () => ({
+  ...jest.requireActual('@wings-software/uicore'),
+  useToaster: jest.fn(() => ({ showError, showSuccess, clear: toasterClear }))
+}))
+
+window.IntersectionObserver = mockIntersectionObserver
+
+// End of mocks
+
+function RenderComponent(props: { initialValues: any }): JSX.Element {
+  const { initialValues } = props || {}
+
+  return (
+    <TestWrapper
+      path={TEST_PATH}
+      pathParams={{
+        accountId: 'testAcc',
+        orgIdentifier: 'testOrg',
+        projectIdentifier: 'test',
+        pipelineIdentifier: 'editPipeline',
+        module: 'cf'
+      }}
+      defaultAppStoreValues={defaultAppStoreValues}
+    >
+      <PipelineContext.Provider value={getPipelineContext()}>
+        <PipelineCanvas {...pipelineProps} />
+        <StageOverview>
+          <Formik initialValues={initialValues} onSubmit={() => undefined} formName="TestWrapper">
+            {formikProps => (
+              <FormikForm>
+                <NameIdDescriptionTags formikProps={formikProps} />
+              </FormikForm>
+            )}
+          </Formik>
+        </StageOverview>
+      </PipelineContext.Provider>
+    </TestWrapper>
+  )
+}
+
+describe('Stage Overview Tests', () => {
+  beforeEach(() => {
+    // eslint-disable-next-line
+    // @ts-ignore
+    useParams.mockImplementation(() => {
+      return { pipelineIdentifier: DefaultNewPipelineId }
+    })
+    // eslint-disable-next-line
+    // @ts-ignore
+    useMutateAsGet.mockImplementation(() => {
+      return mockPipelineTemplateYaml
+    })
+  })
+
+  test(`it displays the overview tab `, async () => {
+    render(<RenderComponent initialValues={{}} />)
+    const overviewTab = screen.getByTestId('stageOverview')
+    await waitFor(() => expect(overviewTab).toBeInTheDocument())
+  })
+
+  test(`it displays the Stage Overview Component `, async () => {
+    render(<RenderComponent initialValues={{}} />)
+    const overviewPanel = screen.getByTestId('stageOverviewPanel')
+    await waitFor(() => expect(overviewPanel).toBeInTheDocument())
+  })
+
+  test(`it displays the save button as disabled until edit`, async () => {
+    render(<RenderComponent initialValues={{}} />)
+    act(() => {
+      fireEvent.click(screen.getByText('save'))
+    })
```

Comment:
you can add below to actually check if it's disabled 
```suggestion
    expect(screen.getByText('save')).toBeDisabled()
```

Reply:
this won't work as you are getting the text not the actual element so it won't pick up the props of the button. So we cannot check if its disabled :( This is why I check the promise is called. I can't add a test id onto the button as it is in Pipeline.